sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -144,10 +152,11 @@ case class GetArrayStructFields(
     field: StructField,
     ordinal: Int,
     numFields: Int,
-    containsNull: Boolean) extends UnaryExpression {
+    containsNull: Boolean) extends UnaryExpression with ExtractValue {
 
   override def dataType: DataType = ArrayType(field.dataType, containsNull)
-  override def toString: String = s"$child.${field.name}"
+  override def toString: String = s"$child[$ordinal].${field.name}"
```

Comment:
The javadoc of `GetArrayStructFields` is too simple...

Actually `GetArrayStructFields` means, we have an array, each element of this array is struct type, and for each element, we wanna get one field of it. It's like a batched version of `GetStructField`, just like `persons.map(_.name)` is a batched version of `person.name`.

So `s"$child.${field.name}"` is corrected here.